cluster/src/main/java/org/apache/iotdb/cluster/query/ClusterPlanExecutor.java

```diff
@@ -81,22 +99,172 @@ public QueryDataSet processQuery(PhysicalPlan queryPlan, QueryContext context)
   }
 
   @Override
-  protected List<String> getNodesList(String schemaPattern, int level) {
-    // TODO-Cluster: enable meta queries
-    throw new UnsupportedOperationException("Not implemented");
-    //return metaGroupMember.getNodeList(schemaPattern, level);
+  protected List<String> getNodesList(String schemaPattern, int level)
+      throws IOException, TException, InterruptedException, MetadataException {
+    ConcurrentSkipListSet<String> nodeSet = new ConcurrentSkipListSet<>(
+        MManager.getInstance().getNodesList(schemaPattern, level));
+
+    ExecutorService pool = new ScheduledThreadPoolExecutor(THREAD_POOL_SIZE);
+
+    for (PartitionGroup group : metaGroupMember.getPartitionTable().getLocalGroups()) {
```

Comment:
Local groups are the groups that the local node is in and they do not necessarily cover all groups. For example, if replica number is 2, and there are 4 nodes 1,2,3,4 the local groups of 1 will only be [4,1] and [1,2].

Reply:
That's right. Thanks for your reminding.